pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -190,10 +190,25 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.WaitUntilControlPlaneReady),
 			Dependencies: flow.NewTaskIDs(deployControlPlane),
 		})
+		syncETCDEncryptionConfiguration = g.Add(flow.Task{
+			Name:         "Syncing etcd encryption configuration to ShootState",
```

Comment:
1. Why is the sync performed before the generation of the etcd secret?
1. Why are there so many different tasks in the flow? Can't they combined into a general "sync data" step? This would also work well with @danielfoehrKn's comment.

Reply:
The sync is only intended to fetch already existing etcd encryption configuration from an existing secret so that it doesn't get regenerated in the Generation step. This will be removed later on.